src/libs/merge/kdbmerge.c

```diff
@@ -181,6 +189,147 @@ static int getTotalNonOverlaps (Key * informationKey)
 	       getNonOverlapOnlyBaseConflicts (informationKey);
 }
 
+static void addConflictingKeys (Key * informationKey, KeySet * conflictingKeys, const char * metaName)
+{
+	increaseStatisticalValue (informationKey, metaName);
+
+	KeySet * meta = keyMeta (informationKey);
+
+	Key * conflictsRoot = keyNew (META_ELEKTRA_MERGE_CONFLICT, KEY_END);
+	for (elektraCursor end, it = ksFindHierarchy (meta, conflictsRoot, &end); it < end; ++it)
```

Comment:
Oh I didn't know it was considered internal API. I think it's very handy so I'd be open to document it. Why would you keep it internal?

Reply:
Mostly because it is poorly documented and tested, so if this fixed the function has very good potential to get public.

The only other problem might be overlap with other functions (is the API still minimal after adding it?). I didn't check this enough, though. @lawli3t did you look at ksFindHierarchy?

But even if it doesn't end up in elektra-core, we can still have it another lib.

Reply:
So I've read again through the doxygen documentation for `ksFindHierarchy` and I find it quite understandable. What do you think is missing @markus2330 ? 

Testing-wise, there are quite a few automated unit tests for this function too. I'm not sure what you mean by it is poorly tested.

I support @kodebach in that it should be made public. It contains quite a few optimizations that would be hard to reimplement outside of the core code of Elektra, especially now with copy-on-write.